pkg/api/handlers/compat/images.go

```diff
@@ -238,12 +241,17 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
 	}
 	defer auth.RemoveAuthfile(authfile)
 
+	platformSpecs := strings.Split(query.Platform, "/")
+
 	pullOptions := &libimage.PullOptions{}
 	pullOptions.AuthFilePath = authfile
 	if authConf != nil {
 		pullOptions.Username = authConf.Username
 		pullOptions.Password = authConf.Password
 		pullOptions.IdentityToken = authConf.IdentityToken
+		pullOptions.OS = platformSpecs[0]
+		pullOptions.Architecture = platformSpecs[1]
+		pullOptions.Variant = platformSpecs[2]
```

Comment:
Please see cmd/podman/images/pull.go:140 for a safer handling of Platform parsing.

Reply:
@cdoern Response from IRC question when using platform, `OS`, `OS/Architecture`, and `OS/Architecture/Variant` are all legal values.